docs/design-doc-provider-identity-based-auth.md

```diff
@@ -0,0 +1,571 @@
+CRs and by having different managed resources refer to these `ProviderConfig`
+instances. 
+
+The Crossplane provider establishes an identity for the requests it will issue
+to the Cloud provider in the [managed.ExternalConnecter][managed.ExternalConnecter]'s `Connect`
+implementation. This involves calling the associated authentication functions from
+the Cloud SDK libraries (such as the [AWS SDK for Go][aws-sdk] or the [Azure
+SDK for Go][azure-sdk]) with the supplied configuration and credentials from the
+referenced `ProviderConfig` instance.
+
+Managed resources and `ProviderConfig`s are cluster-scoped, i.e., they do not exist within a
+Kubernetes namespace but rather exist at the global (cluster) scope. This does
+not fit well into a namespace-based multi-tenancy model, where each tenant is
+confined to its own namespace. The cluster scope is shared between all
+namespaces. In the namespace-based multi-tenancy model, the common approach is
+to have Role-Based Access Control ([RBAC]) rules that disallow a tenant from
+accessing API resources that do not reside in its namespace. Another dimension
+to consider here is that all namespaced tenants are serviced by a shared
+Crossplane provider deployment typically running in the `crossplane-system`
+namespace. This shared provider instance (or more precisely, the [Kubernetes
+ServiceAccount][k8s-sa] that the provider's pod uses) is allowed, via RBAC, to `get` the (cluster-scoped)
+`ProviderConfig` resources. If tenant `subjects` (groups, users,
+ServiceAccounts) are allowed to directly `create` managed resources, then we cannot
+constrain them from referring to any `ProviderConfig` (thus to any Cloud
+provider credential set) in the cluster solely using RBAC. This is because:
+1. RBAC rules allow designated verbs (`get`, `list`, `create`, `update`, etc.)
+   on the specified API resources for the specified subjects. If a subject,
+   e.g., a `ServiceAccount`, is allowed to `create` a managed resource, RBAC
+   alone cannot be used to constrain the set of `ProviderConfig`s that can be
+   referenced by the `create`d managed resource.
+1. The tenant subject itself does not directly access the `ProviderConfig` and in turn
+   the Cloud provider credential set referred by the `ProviderConfig`. It's the
+   Crossplane provider's `ServiceAccount` that accesses these resources, and as
+   mentioned above, this ServiceAccount currently serves all tenants. This
+   implies that we cannot isolate Cloud provider credentials among namespaced
+   tenants by only using RBAC rules if we allow tenant subjects to have `edit`
+   access (`create`, `update`, `patch`) to managed resources. Although it's
+   possible to prevent them from reading Cloud provider credentials of other
+   tenants in the cluster via RBAC rules, it's not possible to prevent them from
+   _using_ those credentials solely with RBAC.
+
+As discussed in detail in the [Crossplane Multi-tenancy Guide][xp-mt],
```

Comment:
The `[Crossplane Multi-tenancy Guide]` link is broken, could you please fix these broken links?

Reply:
Unfortunately can't find document link to resolve this.

Reply:
We can use the following here: https://docs.crossplane.io/knowledge-base/guides/multi-tenant/

Reply:
Fixed